### PR TITLE
Port system launch

### PR DESCRIPTION
**Pull request by esteve**

### CODE REVIEW

**Comment by TakaHoribe** [comment]
`system.launch.xml` is not modified.

**Comment by esteve** [comment]
@TakaHoribe sorry, forgot to push the changes. Please check the latest commit (https://github.com/tier4/autoware_launcher.iv.universe/pull/8/commits/39bafd9c9422aeff7cd649a418a56e5db12918ba) Thanks.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -2,24 +2,35 @@ name: Build and test
...
+    branches:
+      - ros2
   push:
     branches:
       - ros2
 
 jobs:
   build-and-test:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (fred-apex-ai) [inline .github/workflows/build_and_test.yml:13]: I suppose these are old changes to CI and could now be removed from this PR?

File: system_launch/package.xml [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 
   <maintainer email="kenji.miyake@tier4.jp">Kenji Miyake</maintainer>
-  <license>Apache 2</license>
+  <license>Apache License 2.0</license>
 
-  <buildtool_depend>catkin</buildtool_depend>
+  <buildtool_depend>ament_cmake_auto</buildtool_depend>
+
+  <test_depend>ament_lint_auto</test_depend>
+  <test_depend>ament_lint_common</test_depend>
```
Comment (fred-apex-ai) [inline system_launch/package.xml:15]: please add `<exec_depend>` for all those packages whose launch file is used in `system.launch.xml`

**Review by fred-apex-ai — CHANGES_REQUESTED** [review]

File: system_launch/package.xml (hunk 2 above)
Comment (fred-apex-ai) [reply system_launch/package.xml:15]: And when it's done, see if you can launch everything successfully

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (esteve) [reply .github/workflows/build_and_test.yml]: These changes are no longer in the latest commit.

**Review by fred-apex-ai — APPROVED** [review]
@esteve lgtm. Can you launch it locally?

**Comment by esteve** [comment]
@fred-apex-ai still working on that

**Comment by fred-apex-ai** [comment]
@esteve ok, just ping me when it's ready for review again

**Comment by esteve** [comment]
I'm debugging CI, but I can't seem to understand why `control_launch` can't be found by `colcon`, even though `colcon list` actually finds it (see https://github.com/tier4/autoware_launcher.iv.universe/pull/8/checks?check_run_id=1443706269)

**Comment by esteve** [comment]
CI now finally passes (at least it gets past the build stage), I'll apply the same changes to the test stage and this should be ready for review.

File: system_launch/launch/system.launch.xml [hunk 3]
```diff
@@ -2,21 +2,22 @@
...
-      <arg name="config_file" value="$(find autoware_error_monitor)/config/autoware_error_monitor.yaml" />
+    <include file="$(find-pkg-share autoware_error_monitor)/launch/autoware_error_monitor.launch.xml">
+      <arg name="config_file" value="$(find-pkg-share autoware_error_monitor)/config/autoware_error_monitor.yaml" />
     </include>
 
     <!-- Emergency Handler -->
-    <include file="$(find emergency_handler)/launch/emergency_handler.launch">
-      <arg name="config_file" value="$(find emergency_handler)/config/emergency_handler.yaml" />
+    <include file="$(find-pkg-share emergency_handler)/launch/emergency_handler.launch.xml">
+      <arg name="config_file" value="$(find-pkg-share emergency_handler)/config/emergency_handler.yaml" />
```
Comment (TakaHoribe) [inline system_launch/launch/system.launch.xml:20]: please replace with `emergency_handler.param.yaml`.

File: system_launch/launch/system.launch.xml [hunk 4]
```diff
@@ -2,21 +2,22 @@
...
+    <!-- <include file="$(find-pkg-share autoware_state_monitor)/launch/autoware_state_monitor.launch.xml">
+      <arg name="config_file" if="$(eval run_mode == 'online')" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.yaml" />
+      <arg name="config_file" if="$(eval run_mode == 'planning_simulation')" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.planning_simulation.yaml" />
+    </include> -->
 
     <!-- Error Monitor -->
-    <include file="$(find autoware_error_monitor)/launch/autoware_error_monitor.launch">
-      <arg name="config_file" value="$(find autoware_error_monitor)/config/autoware_error_monitor.yaml" />
+    <include file="$(find-pkg-share autoware_error_monitor)/launch/autoware_error_monitor.launch.xml">
+      <arg name="config_file" value="$(find-pkg-share autoware_error_monitor)/config/autoware_error_monitor.yaml" />
```
Comment (TakaHoribe) [inline system_launch/launch/system.launch.xml:15]: please replace with `autoware_error_monitor.param.yaml`

File: system_launch/launch/system.launch.xml [hunk 5]
```diff
@@ -2,21 +2,22 @@
...
     <!-- State Monitor -->
-    <include file="$(find autoware_state_monitor)/launch/autoware_state_monitor.launch">
-      <arg name="config_file" if="$(eval run_mode == 'online')" value="$(find autoware_state_monitor)/config/autoware_state_monitor.yaml" />
-      <arg name="config_file" if="$(eval run_mode == 'planning_simulation')" value="$(find autoware_state_monitor)/config/autoware_state_monitor.planning_simulation.yaml" />
-    </include>
+    <push-ros-namespace namespace="/system"/>
+    <!-- <include file="$(find-pkg-share autoware_state_monitor)/launch/autoware_state_monitor.launch.xml">
+      <arg name="config_file" if="$(eval run_mode == 'online')" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.yaml" />
+      <arg name="config_file" if="$(eval run_mode == 'planning_simulation')" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.planning_simulation.yaml" />
+    </include> -->
```
Comment (TakaHoribe) [inline system_launch/launch/system.launch.xml:11]: Any reason you commended out here?

This is going to be ported like below.

```
    <push-ros-namespace namespace="/system"/>
    <include file="$(find-pkg-share autoware_state_monitor)/launch/autoware_state_monitor.launch.xml">
      <arg name="config_file" if="$(eval &quot;'$(var run_mode)'=='online'&quot;)" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.yaml" />
      <arg name="config_file" if="$(eval &quot;'$(var run_mode)'=='planning_simulation'&quot;)" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.planning_simulation.yaml" />
    </include>
```

**Review by TakaHoribe — COMMENTED** [review]
@esteve Invalid files are loaded. Please check the launch file works correctly.

```
ros2 launch system_launch sytem.launch.xml
```
Even though, with the changes I mentioned, still the autoware_state_monitor died. But it is no the problem of this launch file. Should be fixed out of this PR.

File: system_launch/launch/system.launch.xml (hunk 3 above)
Comment (esteve) [reply system_launch/launch/system.launch.xml]: Fixed in 46aa1dd

File: system_launch/launch/system.launch.xml (hunk 4 above)
Comment (esteve) [reply system_launch/launch/system.launch.xml]: Fixed in 46aa1dd

File: system_launch/launch/system.launch.xml (hunk 5 above)
Comment (esteve) [reply system_launch/launch/system.launch.xml]: When I started this PR the `autoware_state_monitor` package hadn't been ported yet. Re-added in 624243f63221ceea313784cfeed5a27f282a6801

File: system_launch/launch/system.launch.xml [hunk 6]
```diff
@@ -2,21 +2,22 @@
...
-  <group ns="/system">
+  <group>
     <!-- State Monitor -->
-    <include file="$(find autoware_state_monitor)/launch/autoware_state_monitor.launch">
-      <arg name="config_file" if="$(eval run_mode == 'online')" value="$(find autoware_state_monitor)/config/autoware_state_monitor.yaml" />
-      <arg name="config_file" if="$(eval run_mode == 'planning_simulation')" value="$(find autoware_state_monitor)/config/autoware_state_monitor.planning_simulation.yaml" />
+    <push-ros-namespace namespace="/system"/>
+    <include file="$(find-pkg-share autoware_state_monitor)/launch/autoware_state_monitor.launch.xml">
+      <arg name="config_file" if="$(eval "'$(var run_mode)'=='online'")" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.yaml" />
+      <arg name="config_file" if="$(eval "'$(var run_mode)'=='planning_simulation'")" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.planning_simulation.yaml" />
```
Comment (TakaHoribe) [inline system_launch/launch/system.launch.xml:10]:
```suggestion
      <arg name="config_file" if="$(eval &quot;'$(var run_mode)'=='online'&quot;)" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.yaml" />
      <arg name="config_file" if="$(eval &quot;'$(var run_mode)'=='planning_simulation'&quot;)" value="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.planning_simulation.yaml" />
```

**Review by TakaHoribe — COMMENTED** [review]
@esteve The launch is failed due to the invalid syntax. The `"` is nested twice.

**Comment by esteve** [comment]
@TakaHoribe thanks for the second review, I thought the `quot` syntax was not valid in this context. Fixed in https://github.com/tier4/autoware_launcher.iv.universe/pull/8/commits/5f13b3975a0464a1fd0353bba2373988aea193bf

**Pull request merged**